pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -175,17 +176,40 @@ func (c *Controller) Apply(change *sink.Change) error {
 		}
 	}
 
-	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
-	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
 	c.configStoreMu.Unlock()
 
-	if descriptor.Type == model.ServiceEntry.Type {
-		c.serviceEntryEvents(innerStore, prevStore)
+	if descriptor.Type == model.ServiceEntry.Type &&
+		descriptor.Collection == model.ServiceEntry.Collection {
+		for ns, byName := range innerStore {
+			for name, config := range byName {
+				endpoints := []*model.IstioEndpoint{}
+				if se, ok := config.Spec.(*networking.ServiceEntry); ok {
```

Comment:
I assume we only want to process ServiceEntry, correct? If so, then it seems that if `ok == false` here, we'll still drop down and call `EDSUpdate` below ... is that what we want?

Reply:
You are right, the `EDSUpdate` is meant to be inside the ok condition, I will update.